### PR TITLE
Using deprecated CUDA_ARCHITECTURE values now produces an error.

### LINKED ISSUE #319: Remove support for `ALL` and `""` from CMAKE_CUDA_ARCHITECTURES
**Is your feature request related to a problem? Please describe.**
With the deprecation of  `ALL` and `""` from rapids-cmake starting in 23.02 we should remove them in 23.06

### PR DESCRIPTION
**Pull request by robertmaynard**
## Description
rapids-cmake 23.06 deprecated `ALL` and `""` magic values, this ensures that they are a hard error when used instead of a warning.

Fixes #319 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rapids-cmake/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The `cmake-format.json` is up to date with these changes.

### CODE REVIEW

File: testing/cuda/init_arch-all-deprecated.cmake [hunk 1]
```diff
@@ -16,7 +16,6 @@
 include(${rapids-cmake-dir}/cuda/init_architectures.cmake)
 
 
-set(CMAKE_ERROR_DEPRECATED ON)
```
Comment (vyasr) [inline testing/cuda/init_arch-all-deprecated.cmake:19]: Is it worth updating `add_cmake_test` to always set this variable when building all tests? Do we always want to write tests such that deprecations are caught as errors?

**Review by vyasr — APPROVED** [review]
A couple of small questions/suggestions, not sure if we'll make any changes for them.

File: rapids-cmake/cuda/detail/architectures_policy.cmake [hunk 2]
```diff
@@ -21,8 +21,7 @@ rapids_cuda_architectures_policy
 
 .. versionadded:: v23.02.00
 
-Maps deprecated mode values to new supported values and outputs rapids-cmake
-deprecation warnings.
+Errors on deprecated mode values and outputs guidance on how to upgrade
```
[the remark this answers is not in the document]
Comment (robertmaynard) [reply rapids-cmake/cuda/detail/architectures_policy.cmake]: Updated

**Pull request merged**